### ISSUE TITLE
environment variables list?

### ISSUE THREAD

**Issue by AlecTroemel** [mehcode/config-rs#117]
is it possible to parse an env var into something like `Vec<String>`? I get this error for any formatting I try:

```
thread 'main' panicked at 'somethings wrong with the settings: invalid type: string "1,2,3", expected a sequence', src/libcore/result.rs:999:5
```

the error to me suggest that it is possible, but I could just be reading things wrong.

**Comment by hbobenicio** [comment]
One could parse an env var string into a `Vec<String>` in multiple ways. Comma Separated Values are the most usual way to encode multiple values into a environment variable. If this feature would ever be picked for implementation it's important to note that an API to define which is the separator character is really desirable. Consider those cases that the user environment use a `;` as the separator (e.g.: Windows).

**Comment by matthiasbeyer** [comment]
I'd rather have an option to parse a settings from `String -> Value` myself, as a user of the crate, than making the crate decide which parsing mechanisms to use.
The crate could provide parsing implementations for simple cases (but I guess that's not even needed if the API around this is built correctly), but ultimatively, the user should have the option to implement parsing themselves.

**Comment by veloxl** [comment]
The changes in [#255] work perfectly for a particular use case I have. When could we expect to bump the crate version to include this feature?

**Comment by matthiasbeyer** [comment]
Hm. The last release was a rather big one and I certainly want to go for smaller releases, but I think it is a bit too early right now. lets see what the next two weeks bring :thinking:

**Comment by veloxl** [comment]
Makes sense, thanks @matthiasbeyer

**Comment by AshleySchaeffer** [comment]
@matthiasbeyer may I politely nag for this to be pushed to the latest crate version. Just hit a need for it.

You may politely tell me to go away :)

**Comment by AshleySchaeffer** [comment]
Just realised I'd prefer to use a HashSet<String> rather a Vec<String> as the output, so maybe this exact feature won't fit the bill for me, but it'd still be a bit easier to do the conversion.

**Comment by joeky888** [comment]
> is it possible to parse an env var into something like Vec<String>? I get this error for any formatting I try:
> thread 'main' panicked at 'somethings wrong with the settings: invalid type: string "1,2,3", expected a sequence', src/libcore/result.rs:999:5

tl;dr:

```rust
Environment::with_prefix("APP")
                    .separator("_")
                    .list_separator(",")
                    .try_parsing(true)
```

**Comment by smfx** [comment]
@joeky888 Thank you! What if sometimes my vector consists of only one element? I tried, for example:

`ACCOUNTS=27778901,`

With error:

`Panic! panicked at 'Can't load configuration: invalid type: string "", expected an integer for key...`

And:

`ACCOUNTS=27778901`

With following error:

`Panic! panicked at 'Can't load configuration: invalid type: integer ... expected a sequence`
